### DISCUSSION TITLE
Variable type lost when defining outputs for a job

### DISCUSSION THREAD

**Discussion by sr1dar** [community/community#27921] [Actions] [answered]
Hello,
Strings get lost when I try to use a variable from previous Job:

Job 1:

I run a python script which has the following code:
```
  import os,json
  env_file = os.getenv('GITHUB_ENV')
  with open(env_file, "a") as myfile:
  myfile.write("MY_VAR="+json.dumps(output_list))
```
I need this variable in another Job, Followed (https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs)
![image](https://user-images.githubusercontent.com/100928638/182199649-fd4ebc97-b3c4-4865-8a80-1c5fdeb12df5.png)

Job 2:

Printing the above variable I set:

![image](https://user-images.githubusercontent.com/100928638/182197997-0bd090cf-e226-4b64-be8d-160c3e5d7d13.png)

The strings get converted into variables 
As a result when I try to print the list in a python shell, Get the following error:

![image](https://user-images.githubusercontent.com/100928638/182198393-305cc229-7f38-4153-afba-3f7172dd659b.png)

**Comment by airtower-luna** [comment, marked as answer]
That's because you run the variable through the shell without escaping in the `echo` command. The shell then interprets the `"` as shell syntax, and in the end it's gone from the output.

For the job that creates the output, you could simply use the `set-output` command directly from Python, so it doesn't go through the shell:

```py
import json
# set output_list somehow
print(f'::set-output name=test::{json.dumps(output_list)}')
```

And you even get a simpler workflow as a bonus. :smile_cat: 

For the receiving job it depends what you actually want to do with it. If you need to insert it directly into a shell command (and not via env or something), you could even run the JSON through [`shlex.quote()`](https://docs.python.org/3/library/shlex.html#shlex.quote) before writing it to the output.

**Reply by sr1dar** [reply]
Does the job! I didn't know I could use `set-output` in python, Thank you so much!
